actors/builtin/init/init_actor.go

```diff
@@ -16,44 +14,18 @@ import (
 
 type Runtime = vmr.Runtime
 
-var AssertMsg = autil.AssertMsg
-
-type InitActorState struct {
-	// responsible for create new actors
-	AddressMap  map[addr.Address]abi.ActorID
-	NextID      abi.ActorID
-	NetworkName string
-}
-
-func (s *InitActorState) ResolveAddress(address addr.Address) addr.Address {
-	actorID, ok := s.AddressMap[address]
-	if ok {
-		idAddr, err := addr.NewIDAddress(uint64(actorID))
-		autil.Assert(err == nil)
-		return idAddr
-	}
-	return address
-}
-
-func (s *InitActorState) MapAddressToNewID(address addr.Address) addr.Address {
-	actorID := s.NextID
-	s.NextID++
-	s.AddressMap[address] = actorID
-	idAddr, err := addr.NewIDAddress(uint64(actorID))
-	autil.Assert(err == nil)
-	return idAddr
-}
-
+// The init actor uniquely has the power to create new actors.
+// It maintains a table resolving pubkey and temporary actor addresses to the canonical ID-addresses.
 type InitActor struct{}
 
 func (a *InitActor) Constructor(rt Runtime) *adt.EmptyValue {
```

Comment:
To stick with the pattern used in other actors, this should accept a second param `_ *adt.EmptyValue`. Unless the init actor is a special case?